test/e2e/common/node/container_probe.go

```diff
@@ -728,6 +728,32 @@ done
 			return false, nil
 		}, 1*time.Minute, framework.Poll).ShouldNot(gomega.BeTrue(), "should not see liveness probes")
 	})
+
+	/*
+		Testname: Pod http liveness probe, disconnected while receiving body
+		Description: A Pod is created with liveness probe that is disconnected unexpectedly while receiving the body.
+		             The pod is never restarted because the error is ignored (issue #122591).
+
+	*/
+	ginkgo.It("should *not* be restarted with a liveness probe that is disconnected unexpectedly", func(ctx context.Context) {
```

Comment:
this is the current behavior, right?

Reply:
As far as I tried, this is the current behavior.

In this case, `unexpected EOF` error is returned from `ReadAtMost()`. Then, both  `Failure` result and this error are returned:
https://github.com/kubernetes/kubernetes/blob/fb7bbd2e5df36212ca5133bcbaa46dcdb4d18f0e/pkg/probe/http/http.go#L102-L109

The error is checked prior to the failure. So, this failure is ignored:
https://github.com/kubernetes/kubernetes/blob/fb7bbd2e5df36212ca5133bcbaa46dcdb4d18f0e/pkg/kubelet/prober/worker.go#L287-L291